metadata/reader.go

```diff
@@ -671,3 +686,13 @@ func (r *reader) NumOfNodes() (i int, _ error) {
 	}
 	return
 }
+
+// partition partitions a slice of type T into chunks of size chunkSize
+func partition[T any](data []T, chunkSize int) [][]T {
```

Comment:
Can we add some unit tests that show this is chunking correctly? Also we should either add a unit test or documentation that chunkSize never be zero or this function will segfault.

Reply:
Sure, I can add unit tests. This was taken from the Go Wiki: https://go.dev/wiki/SliceTricks#batching-with-minimal-allocation.